vendor_data.py

```diff
@@ -244,19 +279,25 @@ def ModExcel(self, Key, Index, NewValue):
                 raise ValueError("Invalid key, run 'VendorData.Keys' to see options.")
             if not Index in VendorDict[Key]:
                 raise ValueError("Invalid index, run 'self.ReadExcel()' to see options")
-            if Key == "E-mail": #validate e-mail address if a value for the key E-mail is modified
+            if Key == "E-mail":
                 CheckVendorMail(NewValue)
-            if Key == "Word Rate": #validate word rate if a value for the key Word Rate is modified
+                self.VendorMail = NewValue
+            elif Key == "Word Rate":
                 CheckWordRate(NewValue)
-            if Key == "CAT Tool": #validate CAT Tool if a value for the key CAT Tool is modified
+                self.WordRate = NewValue
+            elif Key == "CAT Tool":
                 CheckCatTool(NewValue)
-            if Key == "Status": #validate Status if a value for the key Status is modified
+                self.CatTool = NewValue
+            elif Key == "Status":
                 if not NewValue in VendorData.Statuses:
                     raise ValueError("Invalid status, run 'VendorData.Statuses' to see options")
-            ChangeDictionaryValue(VendorDict, Key, Index, NewValue) #change the dictionary value in the dictionary `VendorDict` using the arguments provided for the method ModExcel
-            df = pd.DataFrame(VendorDict) #turn VendorDict dictionary back into a panda DataFrame
+                else:
+                    self.Status = NewValue
+            ChangeDictionaryValue(VendorDict, Key, Index, NewValue)
+            df = pd.DataFrame(VendorDict)
             with pd.ExcelWriter(FileName, engine="openpyxl", mode="w") as writer:
-                df.to_excel(writer, sheet_name = "VendorData", index=False) #overwrite the existing excel file
+                df.to_excel(writer, sheet_name = "VendorData", index=False)
+            return("The vendor was correctly modified.")
         else:

```

Comment:
Very good!

You can also change the `if os.path.exists('...')` to `if not os.path.exists('...')`, throw the error of line 302 in that case, and then move on with the rest of the validation (without an `else`). That way you don't have so much indentation in lines 280 to 300.
But it's just an aesthetic thing, this is also perfectly fine.